ingester/ingester.go

```diff
@@ -213,146 +141,84 @@ func (fi *FirehoseIngester) Start(ctx context.Context) error {
-
-	time, err := bluesky.ParseTime(evt.Time)
-	if err != nil {
-		return fmt.Errorf("parsing timestamp: %w", err)
-	}
-	rr, err := repo.ReadRepoFromCar(ctx, bytes.NewReader(evt.Blocks))
-	if err != nil {
-		return fmt.Errorf("reading repo from car: %w", err)
-	}
-
-	for _, op := range evt.Ops {
-		uri := fmt.Sprintf("at://%s/%s", evt.Repo, op.Path)
-		switch repomgr.EventKind(op.Action) {
-		case repomgr.EvtKindCreateRecord:
-			recordCid, record, err := rr.GetRecord(ctx, op.Path)
-			if err != nil {
-				if errors.Is(err, lexutil.ErrUnrecognizedType) {
-					continue
-				}
-				return fmt.Errorf("create (%s): getting record for op: %w", uri, err)
-			}
-
-			// Ensure there isn't a mismatch between the reference and the found
-			// object.
-			if lexutil.LexLink(recordCid) != *op.Cid {
-				return fmt.Errorf("create (%s): mismatch in record and op cid: %s != %s", uri, recordCid, *op.Cid)
-			}
-
-			if err := fi.handleRecordCreate(
-				ctx, evt.Repo, uri, record,
-			); err != nil {
-				return fmt.Errorf("create (%s): handling record create: %w", uri, err)
-			}
-		case repomgr.EvtKindUpdateRecord:
-			recordCid, record, err := rr.GetRecord(ctx, op.Path)
-			if err != nil {
-				if errors.Is(err, lexutil.ErrUnrecognizedType) {
-					continue
-				}
-				return fmt.Errorf("update (%s): getting record for op: %w", uri, err)
-			}
-
-			// Ensure there isn't a mismatch between the reference and the found
-			// object.
-			if lexutil.LexLink(recordCid) != *op.Cid {
-				return fmt.Errorf("update (%s): mismatch in record and op cid: %s != %s", uri, recordCid, *op.Cid)
-			}
-
-			if err := fi.handleRecordUpdate(
-				ctx, evt.Repo, evt.Rev, uri, time, record,
-			); err != nil {
-				return fmt.Errorf("update (%s): handling record update: %w", uri, err)
-			}
-		case repomgr.EvtKindDeleteRecord:
-			if err := fi.handleRecordDelete(
-				ctx, evt.Repo, uri,
-			); err != nil {
-				return fmt.Errorf("handling record delete: %w", err)
-			}
+	span.SetAttributes(actorDIDAttr(evt.Did))
+
+	evtTime := time.UnixMicro(evt.TimeUS)
+	uri := fmt.Sprintf(
+		"at://%s/%s/%s",
+		evt.Did, evt.Commit.Collection, evt.Commit.RKey,
+	)
+
+	switch evt.Commit.Operation {
+	case models.CommitOperationCreate:
+		if err := fi.handleRecordCreate(
+			ctx,
+			evt.Did,
+			uri,
+			evt.Commit.Collection,
+			evt.Commit.Record,
+		); err != nil {
+			return fmt.Errorf("create (%s): handling record create: %w", uri, err)
+		}
+	case models.CommitOperationUpdate:
+		if err := fi.handleRecordUpdate(
+			ctx,
+			evt.Did,
+			evt.Commit.Rev,
+			uri,
+			evtTime,
+			evt.Commit.Collection,
+			evt.Commit.Record,
+		); err != nil {
+			return fmt.Errorf("update (%s): handling record update: %w", uri, err)
 		}
+	case models.CommitOperationDelete:
+		if err := fi.handleRecordDelete(
+			ctx, evt.Did, uri,
+		); err != nil {
+			return fmt.Errorf("handling record delete: %w", err)
+		}
+	default:
+		fi.log.Warn("unknown commit operation", zap.String("kind", evt.Kind))
```

Comment:
**question (non-blocking):** i wonder if we could use something like sentry for this, so we don’t log and forget?

(does GCP have something integrated? otherwise i have a sentry-compatible sever running, which we could reuse)